File: solutions/q392/q392_solution.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        if len(s) == 0:
            return True
        if len(t) == 0:
            return False

        sIndex, tIndex = 0, 0

        while tIndex < len(t) and sIndex < len(s):
            if(s[sIndex] == t[tIndex]):
                sIndex += 1
            tIndex += 1
        
        return sIndex == len(s)

File: solutions/q392/q392_solution.cs
public class Solution {
    public bool IsSubsequence(string s, string t) {
        if (s.Length == 0) return true;
        if (t.Length == 0) return false;

        int sIndex = 0, tIndex = 0;

        while (tIndex < t.Length && sIndex < s.Length) {
            if (s[sIndex] == t[tIndex]) {
                sIndex++;
            }
            tIndex++;
        }

        return sIndex == s.Length;
    }
}
